presto-influx/src/main/java/io/prestosql/plugin/influx/InfluxHttp.java

```diff
@@ -0,0 +1,152 @@
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Scanner;
+
+/* Using raw HTTP because the influx java library has dependency conflicts and puts the burden of quoting identifiers on the caller */
+public class InfluxHttp
+{
+    private final String host;
+    private final int port;
+    private final boolean useHttps;
+    private final String database;
+    private final String username;
+    private final String password;
+
+    public InfluxHttp(String host, int port, boolean useHttps, String database, String username, String password)
+    {
+        this.host = host;
+        this.port = port;
+        this.useHttps = useHttps;
+        this.database = database;
+        this.username = username;
+        this.password = password;
+    }
+
+    public void execute(String ddl)
+    {
+        try {
+            URL url = getUrl("query",
+                    new ImmutableMap.Builder<String, String>()
+                            .put("q", ddl)
+                            .build());
+            checkSuccess((HttpURLConnection) url.openConnection());
+        }
+        catch (Throwable t) {
+            InfluxError.EXTERNAL.fail(t);
+        }
+    }
+
+    public JsonNode query(String query)
+    {
+        final JsonNode response;
+        try {
+            response = new ObjectMapper().readTree(getUrl("query",
+                    new ImmutableMap.Builder<String, String>()
+                            .put("db", database)
+                            .put("q", query)
+                            .build()));
+        }
+        catch (Throwable t) {
+            InfluxError.EXTERNAL.fail(t);
+            return null;
+        }
+        JsonNode results = response.get("results");
+        InfluxError.GENERAL.check(results != null && results.size() == 1, "expecting one result", query);
+        JsonNode result = results.get(0);
+        if (result.has("error")) {
+            InfluxError.GENERAL.fail(result.get("error").asText(), query);
+        }
+        return result.get("series");
+    }
+
+    public void write(String retentionPolicy, String... lines)
+    {
+        byte[] body = String.join("\n", lines).getBytes(StandardCharsets.UTF_8);
+        try {
+            URL url = getUrl("write",
+                    new ImmutableMap.Builder<String, String>()
+                            .put("db", database)
+                            .put("rp", retentionPolicy)
+                            .build());
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
```

Comment:
Let's avoid abbreviations. 
```suggestion
            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```